packages/amplify_auth_cognito/ios/Classes/FlutterConfirmUserAttributeRequest.swift

```diff
@@ -0,0 +1,40 @@
+
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import Foundation
+import Amplify
+import amplify_core
+
+struct FlutterConfirmUserAttributeRequest {
+    var userAttributeKey: AuthUserAttributeKey
+    var confirmationCode: String
+    
+    init(dict: NSMutableDictionary) {
+        self.userAttributeKey = createAuthUserAttributeKey(keyName: dict["userAttributeKey"] as! String)
+        self.confirmationCode = dict["confirmationCode"] as! String
+    }
+    
+    static func validate(dict: NSMutableDictionary) throws {
+        let validationErrorMessage = "ConfirmUserAttribute Request malformed."
+        if (dict["userAttributeKey"] == nil) {
```

Comment:
I had a similar thought initially. We are checking that the data from the method call is of type `NSMutableDictionary` on every request. I think that is sufficiently performing a null check.

https://github.com/aws-amplify/amplify-flutter/blob/6bc64a7b2bd4f22944b4147d6aa65a6d533663d4/packages/amplify_auth_cognito/ios/Classes/SwiftAuthCognito.swift#L55-L61

It is a little confusing because we *do* have a null check on Android.  I think the difference is that on Android we are checking that the dat is of type `HashMap<*, *>` (see below), which can be null, where `NSMutableDictionary` cannot.

https://github.com/aws-amplify/amplify-flutter/blob/6bc64a7b2bd4f22944b4147d6aa65a6d533663d4/packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/AuthCognito.kt#L120-L125

What I could not figure out, is why we are checking that the data is of type `HashMap<String, *>` on Android in the many of the validate methods. (example below). That seems unnecessary to me since we have already checked that it is of type `HashMap<String, *>`, and then cast it to `HashMap<String, Any>`. I intended to follow that in this PR for the sake on consistency, but I am now seeing that we are already inconsistently checking in Auth (roughly have of the validate methods check for it and half do not). I am going to look into it a bit more, but will plan to remove it from the validate methods I added in this PR unless I find a reason to keep it.

https://github.com/aws-amplify/amplify-flutter/blob/6bc64a7b2bd4f22944b4147d6aa65a6d533663d4/packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/types/FlutterSignInRequest.kt#L42

Reply:
ok thanks for explanation